libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -926,8 +926,12 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         // Telemetry uses this field to determine what type of event should be triggered.
         val setRoutesInfo = when {
             routes.isEmpty() -> SetRoutes.CleanUp
-            routes.first() == directionsSession.routes.firstOrNull() ->
+            routes.first() == directionsSession.routes.firstOrNull() -> {
                 SetRoutes.Alternatives(initialLegIndex)
+            }
+            directionsSession.routes.map { it.id }.contains(routes.first().id) -> {
```

Comment:
The result of execution of this branch relies on the fact that if it's just the alternatives being reordered, we won't go to this branch (because we'll stop on the previous one).
I'm not a fan of this approach, because the result here depends on the branch order, and here it's not super obvious that the order matters. With `if-else` it's more clear than with `when`, although that could be just a personal preference.
Anyway, if anyone else from you finds this point valid, let's maybe make it more explicit here. @DzmitryFomchyn @VysotskiVadim